sim/paladin/guardian_of_ancient_kings.go

```diff
@@ -0,0 +1,226 @@
+		},
+	})
+
+	return paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86150},
+		Flags:          core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskGuardianOfAncientKings,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+			CD: core.Cooldown{
+				Timer:    paladin.NewTimer(),
+				Duration: time.Minute * 5,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, unit *core.Unit, spell *core.Spell) {
+			goakAura.Activate(sim)
+			paladin.AncientGuardian.EnableWithTimeout(sim, paladin.AncientGuardian, duration)
+			paladin.AncientGuardian.CancelGCDTimer(sim)
+		},
+	})
+}
+
+func (paladin *Paladin) registerProtectionGuardian(duration time.Duration) *core.Spell {
+	goakAura := paladin.RegisterAura(core.Aura{
+		Label:    "Guardian of Ancient Kings",
+		ActionID: core.ActionID{SpellID: 86659},
+		Duration: duration,
+
+		// TODO: Perhaps refactor this to also be a pet with a channeled cast?
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			paladin.PseudoStats.DamageTakenMultiplier *= 0.5
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			paladin.PseudoStats.DamageTakenMultiplier /= 0.5
+		},
+	})
+
+	return paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86150},
+		Flags:          core.SpellFlagAPL,
+		ClassSpellMask: SpellMaskGuardianOfAncientKings,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				NonEmpty: true,
+			},
+			CD: core.Cooldown{
+				Timer:    paladin.NewTimer(),
+				Duration: time.Minute * 5,
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, unit *core.Unit, spell *core.Spell) {
+			goakAura.Activate(sim)
+		},
+	})
+}
+
+func (paladin *Paladin) registerRetributionGuardian(duration time.Duration) *core.Spell {
+	var strDepByStackCount = map[int32]*stats.StatDependency{}
+
+	for i := 1; i <= 20; i++ {
+		strDepByStackCount[int32(i)] = paladin.NewDynamicMultiplyStat(stats.Strength, 1.0+0.01*float64(i))
+	}
+
+	ancientPower := paladin.RegisterAura(core.Aura{
+		Label:     "Ancient Power",
+		ActionID:  core.ActionID{SpellID: 86700},
+		Duration:  duration + time.Second*1,
+		MaxStacks: 20,
+
+		OnStacksChange: func(aura *core.Aura, sim *core.Simulation, oldStacks int32, newStacks int32) {
+			if oldStacks > 0 {
+				paladin.DisableDynamicStatDep(sim, strDepByStackCount[oldStacks])
+			}
+
+			if newStacks > 0 {
+				paladin.EnableDynamicStatDep(sim, strDepByStackCount[newStacks])
+			}
+		},
+	})
+
+	ancientFuryMinDamage, ancientFuryMaxDamage :=
+		core.CalcScalingSpellEffectVarianceMinMax(proto.Class_ClassPaladin, 0.2366, 0.3)
+	numTargets := paladin.Env.GetNumTargets()
+	results := make([]*core.SpellResult, numTargets)
+
+	ancientFury := paladin.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 86704},
+		SpellSchool:    core.SpellSchoolHoly,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          core.SpellFlagAPL,
```

Comment:
Is this spell castable manually? Prob dont want the APL flag in here

Reply:
No you're right, I shouldn't write code at 3 in the morning :)